### ISSUE TITLE
Health endpoint improvement

### ISSUE THREAD

**Issue by davidgengenbach** [distribworks/dkron#652]
**Is your feature request related to a problem? Please describe.**
We had a problem with a killed plugin process (due to OOM) which resulted in non-executing jobs.

**Describe the solution you'd like**
The [health endpoint](https://github.com/distribworks/dkron/blob/744911731d0c2c85fec2add4dba6046ead95a9ec/dkron/api.go#L58) might be used to check whether all plugin processes are up and running.
In general, more health checks would be helpful, e.g. cluster health?

**Comment by victorcoder** [comment]
Already on the roadmap, will work on this.

**Comment by victorcoder** [comment]
@davidgengenbach not really the improvement you mention but I think it's better to fail fast in case of a missing plugin. In case of using as a service the OS supervisor will take care of restarting. This is the case with processor plugins.

**Comment by davidgengenbach** [comment]
@victorcoder Yes, I would agree. Having non-functional (= killed) plugins should really result in a killed main-process.

The linked PR will only kill the main process when a job has been executed unsuccessfully (e.g. by a plugin error) which may be far later than the actual plugin process exiting - a periodic health check could circumvent this.
